### ISSUE TITLE
Size of holes to fill

### ISSUE THREAD

**Issue by oleg-alexandrov** [CGAL/cgal#4410]
## Issue Details

In https://doc.cgal.org/latest/Polygon_mesh_processing/index.html it is described how to fill holes. 

Unfortunately, not all meshes are closed. This algorithm will try to fill any hole, even one at the outer boundary of a mesh. Imagine here a mesh that represents a portion of the terrain surface, such a mesh has a boundary where it terminates, and it also may have some holes.

In such a situation the algorithm hangs, and likely will give a junk result.

There is likely no fool-proof solution here. At least, the algorithm could choose not to fill holes whose boundary have more than a given number of edges, or a hole with given diameter. 

# Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux
* Compiler: gcc 
* Release or debug mode:
* Specific flags used (if any):
* CGAL version: 5
* Boost version:
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by sloriot** [comment]
The algorithm is working on boundary cycles that the user provides as input. So you are free to measure the size of the boundary cycle (#edges or length) before trying to fill it.

**Comment by MaelRL** [comment]
There is also some code that does just that in the following file: https://github.com/CGAL/cgal/blob/master/Polyhedron/demo/Polyhedron/Plugins/PMP/Hole_filling_plugin.cpp as the CGAL demo already has a button "Select small holes".

**Comment by oleg-alexandrov** [comment]
Thank you. That example at 

https://doc.cgal.org/latest/Polygon_mesh_processing/Polygon_mesh_processing_2hole_filling_example_8cpp-example.html

is still however not very helpful. It does not explain how to get a boundary cycle that has this particular edge and how one would find its number of edges. I guess it is simple if you already know that, and it could be simple for you to add that, but as  a new user I find this very intimidating.

In short, it could be little work for this example to be improved as the user does not get sufficient knowledge of how a hole could be understood before deciding to fill it or not. And for me the whole code just hanged, I guess when it got to a huge hole when it would give junk results anyway.
